### PR TITLE
fix: change field style colors when input is not editable

### PR DESCRIPTION
**Pull request by himanshuragi456**
### Details

A separate(dull/disabled-looking) style will be provided to fields that are not editable, to provide clear understanding to the user that the field is disabled.

### Fixed Issues

$ https://github.com/Expensify/App/issues/25577
PROPOSAL: https://github.com/Expensify/App/issues/25577#issuecomment-1686245385

### Tests

1. Go to Settings > Workspaces
2. Select a workspace and access "Connect Bank Account."
3. Click Connect Manually
4. Enter Valid Routing Number & Account Number and click continue
5. Fill all required Details and click continue
6. Return to the Connect bank account(Step 1).
7. Observe disabled/dull colors of fields that are not editable.

- [x] Verify that no errors appear in the JS console

### Offline tests

same as online steps

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-09-01 at 11 14 35 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-08-30 at 4 15 22 AM

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-09-06 at 12 22 19 AM

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-09-01 at 11 14 35 PM

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-09-11 at 6 08 37 PM

</details>

<details>
<summary>Android</summary>

Screenshot 2023-08-30 at 4 14 53 AM

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
Hmm I don't know that we should use the disabled cursor there. They aren't technically disabled inputs, they are just read-only text. So I think I would opt to do nothing here.

**Comment by shawnborton** [comment]
Looks good, thanks!

File: src/components/TextInput/BaseTextInput.js
Comment (hayata-suenaga) [inline src/components/TextInput/BaseTextInput.js:268]: if you have time, could provide screenshots that shows the difference when we have this line and not?

Comment (shawnborton) [reply src/components/TextInput/BaseTextInput.js:268]: Rather than messing with the text input component, is there a way to reuse the global styles we're using for this elsewhere, like the profile page?

image

**Comment by hayata-suenaga** [comment]
@himanshuragi456 I'd like to ask a clarifying question here 🙇 

What issue are we trying to solve now?

> No sign is shown to indicating the account number and routing number can't be changed on Connect bank account

This was the original issue. Now because [we don't have to use disabled cursors](https://github.com/Expensify/App/pull/25991#issuecomment-1695479971) now, what are we tying to do in this PR?

**Comment by shawnborton** [comment]
Good on my end. Thanks!

**Comment by Santhosh-Sellavel** [comment]
@shawnborton This would affect the date picker are we fine with this?

#### Left side New & Right Side old

Screenshot 2023-08-30 at 12 51 30 AM

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 What are the pages that do have uneditable fields, please list them down. So it will be helpful for QA to test them.

**Comment by shawnborton** [comment]
We should not be touching the datepicker dropdown. That's why I suggested
to not use a disabled input for this, we should just make it read-only text
instead.

**Comment by Santhosh-Sellavel** [comment]
@shawnborton 
> We should not be touching the datepicker dropdown. That's why I suggested to not use a disabled input for this, we should just make it read-only text instead.

I think it makes sense to make it read-only text. Only on the page where it's meant to be read-only Ex: Profile page. There are flows where it will be both editable & disabled. There we just continue using TextInput and just disable it whenever required. If we try to make it a read-only pattern while temporarily disabled that would be over-engineering. 

In this case, we just add a disabled cursor style or just do nothing!

cc: @hayata-suenaga

**Comment by shawnborton** [comment]
Hmm I am not following your suggestion, mind rephrasing?

**Comment by Santhosh-Sellavel** [comment]
[This](https://github.com/Expensify/App/issues/25577) is the issue we are fixing.

Problem: No sign is shown indicating they can't be changed

In this PR we are trying to make those fields read-only when it's disabled.

IMO, We should make text read-only when its purpose is truly read-only. For example, on the Profile page, we use read-only patterns to show the display name, contacts, etc. which are set.

But in the present issue, that's not the case. Connect Bank Manual Step page where we edit the field in the initial connection flow and show it disabled in after we partially save data to the server.  So we show it as editable and disabled depends on the state introducing read-only patterns would be confusing and implementation-wise would be over-engineering.
So we should just stick with the current pattern using textinput and keep it disabled when required.

The problem statement being
> No sign is shown indicating they can't be changed

Now we can
- Show the Disabled cursor when the field  is disabled to convey user convincingly

OR

-  Just Do Nothing because we are partially conveying that the field is not editable by not allowing to focus. i.e. if the field is editable we would focus and show the text cursor on the field, we don't do that already.


@shawnborton Do you get it?

cc: @hayata-suenaga

**Comment by shawnborton** [comment]
Got it. If we want to show this still as a text input but in the disabled state, I would recommend we use the far-right column here as our style:
image

This way it is clear that this was an editable input, but it can no longer be edited. That being said, I don't think that kind of style should apply to the datepicker, as we want the datepicker to appear like it's an input that can be interacted with (because it can be).

Let me know if that makes sense. Sorry for the run around here, I thought we could just display the text as read-only and not worry about a disabled input state.

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 Can you make the changes to address this with styles recommended [here](https://github.com/Expensify/App/pull/25991#issuecomment-1698075535)? 

Thanks!

**Comment by Santhosh-Sellavel** [comment]
> do we have that in use somewhere already? where can I get the exact color variables?

https://github.com/Expensify/App/blob/main/src/styles/colors.js

**Comment by Santhosh-Sellavel** [comment]
> also, please help me find the datepicker.

Not sure why you would need that. Are you asking about how to find in UI or in code?

File: src/components/TextInput/BaseTextInput.js
Comment (Santhosh-Sellavel) [inline src/components/TextInput/BaseTextInput.js:269]: I think we should use `disabled` prop instead of `isEditable`

File: src/components/TextInput/BaseTextInput.js
Comment (Santhosh-Sellavel) [inline src/components/TextInput/BaseTextInput.js:244]: This would be unnecessary

**Comment by shawnborton** [comment]
The value inside of the disabled inputs should use our color.textSupporting color. Can you update? Can you share what the cursor looks like, too? And can you share if this affects the date picker as well?

File: src/components/TextInput/BaseTextInput.js
Comment (Santhosh-Sellavel) [inline src/components/TextInput/BaseTextInput.js:336]: This seems unnecessary

File: src/components/TextInput/BaseTextInput.js
Comment (Santhosh-Sellavel) [inline src/components/TextInput/BaseTextInput.js:244]: Same unnecessary revert this too

**Comment by shawnborton** [comment]
Where can I find updated mockups?

**Comment by shawnborton** [comment]
Ah, can you please update the value color to use text.Supporting please?

**Comment by shawnborton** [comment]
Yes, that's what I mean.

**Comment by shawnborton** [comment]
Much better, thanks!

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 Please update PR details correctly

**Comment by Santhosh-Sellavel** [comment]
> Border-bottom will be removed for fields that are not editable, to provide clear understanding to the user that the field is disabled.

This is not the solution we are implemented here, so update details accordingly

**Review by hayata-suenaga — DISMISSED** [review]
code change looks good

@himanshuragi456 we also changed the background color for disabled text fields right?

could you include a step to confirm that in your test instruction?

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web & Desktop </summary>

https://github.com/Expensify/App/assets/85645967/3e3fd123-4c16-47b6-9764-cb8dbad6acce

</details>

<details>
<summary>Mobile Web - Chrome & Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/85645967/97914dd6-f5de-430b-8a99-794d0b6e7f8a

</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 
mWeb iOS Looks a bit inconsistent from mWeb Android / Web
Screenshot 2023-09-04 at 4 33 34 AM

cc: @shawnborton

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 

> Proceed to Step 3 out of 5.

This is a bit confusing and can be elaborative

- Click Connect Manually
- Enter Valid Routing Number & Account Number and click continue
- Fill all required Details and click continue

**Comment by shawnborton** [comment]
I agree, let's fix that. Perhaps we need to do some kind of `-webkit-appearance: none` to the inputs in the case that Safari is automatically adding styles or something?

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 Please ask for help in #expensify-open-source in slack. Let's see what others has to say
> The color value is the same for both safari and chrome. toggling them also changes the color in UI, which confirms that these are the target properties. Now, in spite of having the same value RGB(175, 187, 176), these are rendering different colors.

**Comment by Santhosh-Sellavel** [comment]
This PR will be on Merge Freeze anyways, Let's look for a solution meanwhile, thanks!

cc: @hayata-suenaga

File: src/styles/styles.js
Comment (Santhosh-Sellavel) [inline src/styles/styles.js:784]: Can you add a comment explaining why we need this?

Also elaborate how this solves our issue thanks!

File: src/components/TextInput/BaseTextInput.js
Comment (Santhosh-Sellavel) [inline src/components/TextInput/BaseTextInput.js:334]: This comment does not make sense anymore

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 Console Warnings resolve this
Simulator Screenshot - iPhone 14 - 2023-09-07 at 01 38 58
Simulator Screenshot - iPhone 14 - 2023-09-07 at 01 36 35

**Comment by Santhosh-Sellavel** [comment]
@himanshuragi456 

Penalties are waived when a delay happens because of something beyond our control. Here we had one such case, i.e. we just deviated from the expected outcome here on the PR.

> It will still take more effort to resolve the console warning.

IMO this is something which within our control, this could have been caught just running through the checklist conciously. 
The console error occurs due to the changes made recently to fix this [issue](https://github.com/Expensify/App/pull/25991#issuecomment-1704429751). It's straightforward not a complex issue and does not take more time either.

`WebkitTextFillColor` style is not available on native platforms, so we should apply this style to Safari Web. We have examples already in the repo.

Please try to complete the requested changes soon, thanks!

cc: @hayata-suenaga

**Comment by hayata-suenaga** [comment]
@himanshuragi456 I answered your payment question on the GH issue.

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web & Desktop </summary>

Screenshot 2023-09-08 at 11 51 10 PM

https://github.com/Expensify/App/assets/85645967/87b1beab-080e-49bb-aed9-6008bbd77468

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-09-08 at 11 59 51 PM

</details>

<details>
<summary>Mobile Web - Safari</summary>

Simulator Screenshot - iPhone 14 - 2023-09-08 at 23 57 56

</details>

<details>
<summary>iOS</summary>

Simulator Screenshot - iPhone 14 - 2023-09-08 at 23 56 47

</details>

<details>
<summary>Android</summary>

Screenshot 2023-09-08 at 11 56 31 PM

</details>

**Review by Santhosh-Sellavel — APPROVED** [review]
@hayata-suenaga Please verify workflow checks are good, otherwise LGTM, tests well!
Also, I think this could wait for Merge Freeze, I let you make the final call though thanks!

cc: @shawnborton Please verify screenshots!

**Comment by shawnborton** [comment]
Screenshots generally look good but I noticed that web has a bottom-border on the inputs but iOS does not. Why is there an inconsistency there?

**Comment by hayata-suenaga** [comment]
@himanshuragi456 could you check this ⬆️

File: src/styles/styles.js
Comment (shawnborton) [reply src/styles/styles.js]: Thanks!

**Comment by shawnborton** [comment]
Everything is good on my end.

**Review by hayata-suenaga — APPROVED** [review]

**Comment by hayata-suenaga** [comment]
sorryyy I missed this one thank you for letting me know 😄

**Pull request merged**
